pyo3-macros-backend/src/method.rs

```diff
@@ -173,10 +184,15 @@ impl SelfType {
                 } else {
                     syn::Ident::new("extract_pyclass_ref", *span)
                 };
+                let holder = syn::Ident::new(&format!("holder_{}", holders.len()), *span);
+                holders.push(quote! {
```

Comment:
I think this fixes the UI test "regression", I assume by causing the new error to be deduplicated away based on the span:

```suggestion
                holders.push(quote_spanned! { *span =>
```

Reply:
Ok it sounds like you solved it 👀

Reply:
Indeed this fixed it. Thanks!

Reply:
I solved the one part which was definitely a regression but this also solves the extra error messages hence is definitely better.